`-Wconf` options are processed incorrectly

## Compiler version

3.3.0

## Minimized code

```Scala
object Main {
  def main(args: Array[String]): Unit = Deprecated.deprecated
}

object Deprecated {
  @deprecated("", "")
  def deprecated = ()
}
```
(anything that will produce a warning that can be configured with `-Wconf` will do)

## Output
scala 2.13.10
```bash
$ scala-2.13.10/bin/scalac -Wconf:cat=deprecation:e -Wconf:cat=deprecation:s Test.scala

$ scala-2.13.10/bin/scalac -Wconf:cat=deprecation:s -Wconf:cat=deprecation:e Test.scala
Test.scala:3: error: method deprecated in object Deprecated is deprecated:
  def main(args: Array[String]): Unit = Deprecated.deprecated
                                                   ^
1 error
```
scala 3.3.0
```bash
$ scala3-3.3.0/bin/scalac -Wconf:cat=deprecation:e -Wconf:cat=deprecation:s Test.scala
-- Error: Test.scala:3:51 -----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
3 |  def main(args: Array[String]): Unit = Deprecated.deprecated
  |                                        ^^^^^^^^^^^^^^^^^^^^^
  |                                        method deprecated in object Deprecated is deprecated since : 
1 error found

$ scala3-3.3.0/bin/scalac -Wconf:cat=deprecation:s -Wconf:cat=deprecation:e Test.scala
```

## Expectation
In scala 2, passing `-Wconf:cat=deprecation:silent` after `-Wconf:cat=deprecation:error` would override the `error` setting, i.e. the last provided configuration wins. In scala 3, the opposite is true: the first provided configuration wins. The documentation for the behavior is the same for both: `scalac -Wconf:help` includes the following:
>User-defined configurations are added to the left. The leftmost rule matching
>a warning message defines the action.

I would expect that, having the same documentation, the -Wconf flags would be handled in the same order. The scala 2 behavior is useful for providing a general default (e.g. `-Wconf:any:error`) and then being able to append more specific options to the scalac options and have those override the general default. A bit more discussion can be found in https://github.com/lampepfl/dotty/discussions/18365

--- a/compiler/src/dotty/tools/dotc/reporting/WConf.scala
+++ b/compiler/src/dotty/tools/dotc/reporting/WConf.scala
@@ -106,7 +106,7 @@ object WConf:
   def fromSettings(settings: List[String]): Either[List[String], WConf] =
     if (settings.isEmpty) Right(WConf(Nil))
     else
-      val parsedConfs: List[Either[List[String], (List[MessageFilter], Action)]] = settings.map(conf =>
+      val parsedConfs: List[Either[List[String], (List[MessageFilter], Action)]] = settings.reverse.map(conf =>
         val filtersAndAction = conf.split(':')
         if filtersAndAction.length != 2 then Left(List("exactly one `:` expected (<filter>&...&<filter>:<action>)"))
         else

--- a/compiler/test/dotty/tools/dotc/config/ScalaSettingsTests.scala
+++ b/compiler/test/dotty/tools/dotc/config/ScalaSettingsTests.scala
@@ -83,4 +83,18 @@ class ScalaSettingsTests:
     val nowr = new Diagnostic.Warning("This is a problem.".toMessage, util.NoSourcePosition)
     assertEquals(Action.Silent, sut.action(nowr))
 
+  @Test def `i18367 rightmost WConf flags take precedence over flags to the left`: Unit =
+    import reporting.{Action, Diagnostic}
+    val sets = new ScalaSettings
+    val args = List("-Wconf:cat=deprecation:e", "-Wconf:cat=deprecation:s")
+    val sumy = ArgsSummary(sets.defaultState, args, errors = Nil, warnings = Nil)
+    val proc = sets.processArguments(sumy, processAll = true, skipped = Nil)
+    val conf = sets.Wconf.valueIn(proc.sstate)
+    val msg  = "Don't use that!".toMessage
+    val depr = new Diagnostic.DeprecationWarning(msg, util.NoSourcePosition)
+    val sut  = reporting.WConf.fromSettings(conf).getOrElse(???)
+    assertEquals(Action.Silent, sut.action(depr))
+
+
+
 end ScalaSettingsTests

